microprofile/opentracing-smallrye/src/main/java/org/wildfly/extension/microprofile/opentracing/smallrye/TracingDependencyProcessor.java

```diff
@@ -0,0 +1,36 @@
+package org.wildfly.extension.microprofile.opentracing.smallrye;
+
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.Phase;
+import org.jboss.as.server.deployment.module.ModuleDependency;
+import org.jboss.as.server.deployment.module.ModuleSpecification;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleIdentifier;
+import org.jboss.modules.ModuleLoader;
+
+public class TracingDependencyProcessor implements DeploymentUnitProcessor {
+  public static final ModuleIdentifier EXTENSION_MODULE = ModuleIdentifier.create("org.wildfly.extension.microprofile.opentracing-smallrye");
+
+  public static final Phase PHASE = Phase.DEPENDENCIES;
+  public static final int PRIORITY = 0x4000;
+
+  @Override
+  public void deploy(DeploymentPhaseContext phaseContext) {
+    DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+    addDependencies(deploymentUnit);
+  }
+
+  private void addDependencies(DeploymentUnit deploymentUnit) {
+    ModuleSpecification moduleSpecification = deploymentUnit.getAttachment(Attachments.MODULE_SPECIFICATION);
+    ModuleLoader moduleLoader = Module.getBootModuleLoader();
+
+    moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, EXTENSION_MODULE, false, true, true, false));
```

Comment:
I think some sort of split up of the org.wildfly.extension.microprofile.opentracing-smallrye module is needed. We shouldn't be exposing subsystem modules to deployments. There are a couple places where we do and they're regrettable.

I commented on the MP Config PR the same way.

Reply:
The first version of this PR had a very granular set of modules, but I ended up having CDI problems (beans registered in one module couldn't be seen/consumed by another module).

Would it be OK to have this as a JIRA to be completed before this gets promoted from tech-preview?